docs_src/design/adr/014-Secret-Provider-For-All.md

```diff
@@ -0,0 +1,409 @@
+  - The standard `SecretClient` is considered the shared client for secrets that all Application Service instances share. It is only used by the `GetDatabaseCredentials` API
+  - Configuration based secret store for non-secure mode called `InsecureSecrets`
+  - Caching of secrets
+    - Needed so that secrets used by pipeline functions do not cause call out to Vault for every Event processed
+
+#### What is a Secret?
+
+A secret is a collection of key/value pairs stored in a `SecretStore` at specified path whose values are sensitive in nature. Redis database credentials are an example of a `Secret` which contains the `username` and `password` key/values stored at the `redis` path.
+
+#### Service Exclusive vs Service Shared Secrets
+
+**Service Exclusive** secrets are those that are exclusive to the instance of the running service. An example of exclusive secrets are the HTTP Auth tokens used by two running instances of app-service-configurable (http-export) which export different device Events to different endpoints with different Auth tokens in the HTTP headers.  Service Exclusive secrets are seeded by POSTing the secrets to the `/api/vX/Secrets` endpoint on the running instance of each Application Service.
+
+**Service Shared** secrets are those that all instances of a class of service, such a Application Services, share. Think of Core Data as it own class of service. An example of shared secrets are the database credentials for the single database instance for Store and Forward data that all Application Services may need to access. Another example is the database credentials for each of instance the Core Data. It is shared, but only one instance of Core Data is currently ever run. Service Shared secrets are seeded by security-secretstore-setup using static configuration for static secrets for known services. Currently database credentials are the only shared secrets. In the future we may have Message Bus credentials as shared secrets, but these will be truly shared secrets for all services to securely connect to the Message Bus, not just shared between instances of a service.
+
+Application Services currently have the ability to configure `SecretStores` for **Service Exclusive** and/or **Service Shared** secrets depending on their needs.
+
+#### Known and Unknown Services 
+
+- **Known Services** are those identified in the static configuration by security-secretstore-setup
+  
+  - These currently are Core Data, Core Metadata, Support Notifications, Support Scheduler and Application Service (class)
+  
+- **Unknown Services** are those not known in the static configuration that become known when added to the Docker compose file or Snap. 
+
+  - Application Service (instance) are examples of these services. 
+
+  - Service exclusive `SecretStore` can be created for these services by adding the services' unique name , i.e. appservice-http-export, to the `ADD_SECRETSTORE_TOKENS` environment variable for security-secretstore-setup
```

Comment:
Should specify which service needs the env var.

Reply:
Already do. `environment variable for security-secretstore-setup`